radixdlt/src/main/java/com/radixdlt/consensus/bft/VertexStore.java

```diff
@@ -492,7 +496,7 @@ public VertexMetadata insertVertex(Vertex vertex) throws VertexInsertionExceptio
 		}
 
 		for (Vertex committed : path) {
-			CommittedAtom committedAtom = new CommittedAtom(committed.getAtom(), commitMetadata);
+			CommittedAtom committedAtom = new CommittedAtom(committed.getAtom(), commitQC);
```

Comment:
Let's add a comment that we will want to remove QC from CommittedAtom at some point and just include data and signatures required to verify atom was committed. Otherwise QC will contain extra info (proposed and parent data) which is not required at the Ledger layer.

Reply:
Not sure I understand this.  A QC is just VoteData plus signatures (and now timestamps).  The VoteData is what is signed in a Vote, so that is required to verify signatures.

Could you elaborate?  Maybe I'm missing something here.

Reply:
Yes, VoteData also contains data for the proposed and the parent vertices, data which should be opaque to the Ledger/StateComputer. If you see Libra, they use a structure called LedgerInfoWithSignatures